mgradm/cmd/backup/restore/restore.go

```diff
@@ -0,0 +1,266 @@
+// SPDX-FileCopyrightText: 2025 SUSE LLC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package restore
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"slices"
+	"strings"
+
+	"github.com/rs/zerolog/log"
+	"github.com/spf13/cobra"
+
+	"github.com/uyuni-project/uyuni-tools/mgradm/cmd/backup/shared"
+	podman_mgradm "github.com/uyuni-project/uyuni-tools/mgradm/shared/podman"
+	. "github.com/uyuni-project/uyuni-tools/shared/l10n"
+	"github.com/uyuni-project/uyuni-tools/shared/podman"
+	"github.com/uyuni-project/uyuni-tools/shared/types"
+	"github.com/uyuni-project/uyuni-tools/shared/utils"
+)
+
+var runCmdInput = utils.RunCmdInput
+var runCmd = utils.RunCmd
+var systemd = podman.SystemdImpl{}
+
+func Restore(
+	_ *types.GlobalFlags,
+	flags *shared.Flagpole,
+	_ *cobra.Command,
+	args []string,
+
+) error {
+	inputDirectory := args[0]
+	printIntro(inputDirectory, flags)
+	dryRun := flags.DryRun
+	// SanityCheck
+	if err := sanityChecks(inputDirectory, flags); err != nil {
+		return shared.AbortError(err, false)
+	}
+
```

Comment:
maybe do we need a `StorageCheck` also here ? I'm wondering to check if it has sense to check if there's enough space in the disk to untar the backup...
Maybe not required since the `restore` would fail and can re-run it after the system was cleaned up